micrometer-core/src/test/java/io/micrometer/core/instrument/binder/kafka/KafkaConsumerMetricsTest.java

```diff
@@ -15,61 +15,145 @@
-        props.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, BOOTSTRAP_SERVERS);
-        props.put(ConsumerConfig.GROUP_ID_CONFIG, "MicrometerTestConsumer");
+        props.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, sharedKafkaTestResource.getKafkaConnectString());
+        props.put(ConsumerConfig.GROUP_ID_CONFIG, groupId);
         props.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, LongDeserializer.class.getName());
         props.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class.getName());
+        props.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+        props.put(ConsumerConfig.CLIENT_ID_CONFIG, clientId);
 
         Consumer<Long, String> consumer = new KafkaConsumer<>(props);
         consumer.subscribe(Collections.singletonList(TOPIC));
+        return consumer;
+    }
+
+    private void sendMessage() throws Exception {
+        Properties props = new Properties();
+        props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, sharedKafkaTestResource.getKafkaConnectString());
+        props.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, LongSerializer.class.getName());
+        props.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName());
+        Producer<Long, String> producer = new KafkaProducer<>(props);
+
+        producer.send(new ProducerRecord<>(TOPIC, "Hello")).get();
+        producer.close();
+
     }
 
     @Test
     void consumerMetrics() {
-        createConsumer();
-        createConsumer();
+        createConsumer("Test", "C1");
+        createConsumer("Test", "C2");
 
         MeterRegistry registry = new SimpleMeterRegistry();
         kafkaConsumerMetrics.bindTo(registry);
 
         // fetch metrics
-        registry.get("kafka.consumer.records.lag.max").tag("client.id", "consumer-1").gauge();
-        registry.get("kafka.consumer.records.lag.max").tag("client.id", "consumer-2").gauge();
+        registry.get("kafka.consumer.records.lag.max").tag("client.id", "C1").gauge();
+        registry.get("kafka.consumer.records.lag.max").tag("client.id", "C2").gauge();
 
         // consumer group metrics
-        registry.get("kafka.consumer.assigned.partitions").tag("client.id", "consumer-1").gauge();
+        registry.get("kafka.consumer.assigned.partitions").tag("client.id", "C1").gauge();
 
         // global connection metrics
-        registry.get("kafka.consumer.connection.count").tag("client.id", "consumer-1").gauge();
+        registry.get("kafka.consumer.connection.count").tag("client.id", "C1").gauge();
     }
 
     @Test
-    void kafkaMajorVersion() {
-        createConsumer();
+    void consumerMetricsWithConsumerStartAfterBinding() {
+
+        MeterRegistry registry = new SimpleMeterRegistry();
+        kafkaConsumerMetrics.bindTo(registry);
 
-        assertThat(kafkaConsumerMetrics.kafkaMajorVersion(Tags.of("client.id", "consumer-1"))).isGreaterThanOrEqualTo(2);
+        createConsumer("Test", "C4");
```

Comment:
Consumer is not closed after execution. Applicable for all tests. 
Please address here or wait for this PR completion - https://github.com/micrometer-metrics/micrometer/pull/1098